Xamarin.Forms.Core/Span.cs

```diff
@@ -1,145 +1,101 @@
-				set { SetValue(FontElement.FontProperty, value); }
-			}
-
-			public FontAttributes FontAttributes {
-				get { return (FontAttributes)GetValue(FontElement.FontAttributesProperty); }
-				set { SetValue(FontElement.FontAttributesProperty, value); }
-			}
-
-			public string FontFamily {
-				get { return (string)GetValue(FontElement.FontFamilyProperty); }
-				set { SetValue(FontElement.FontFamilyProperty, value); }
-			}
-
-			[TypeConverter(typeof(FontSizeConverter))]
-			public double FontSize {
-				get { return (double)GetValue(FontElement.FontSizeProperty); }
-				set { SetValue(FontElement.FontSizeProperty, value); }
-			}
-
-			double IFontElement.FontSizeDefaultValueCreator() =>
-				((IFontElement)_span).FontSizeDefaultValueCreator();
-
-			public void OnFontAttributesChanged(FontAttributes oldValue, FontAttributes newValue) =>
-				((IFontElement)_span).OnFontAttributesChanged(oldValue, newValue);
-
-			public void OnFontChanged(Font oldValue, Font newValue) =>
-				((IFontElement)_span).OnFontChanged(oldValue, newValue);
-
-			public void OnFontFamilyChanged(string oldValue, string newValue) =>
-				((IFontElement)_span).OnFontFamilyChanged(oldValue, newValue);
-
-			public void OnFontSizeChanged(double oldValue, double newValue) =>
-				((IFontElement)_span).OnFontSizeChanged(oldValue, newValue);
-
-			protected override void OnPropertyChanged(string propertyName = null)
-			{
-				base.OnPropertyChanged(propertyName);
-				_span.OnPropertyChanged(propertyName);
-			}
+			_mergedStyle = new MergedStyle(GetType(), this);
 		}
 
-		Color _backgroundColor;
-
-		BindableObject _fontElement;
-
-		Color _foregroundColor;
+		public static readonly BindableProperty StyleProperty = BindableProperty.Create("Style", typeof(Style), typeof(Span), default(Style),
+			propertyChanged: (bindable, oldvalue, newvalue) => ((Span)bindable)._mergedStyle.Style = (Style)newvalue);
 
-		string _text;
-
-		public Span()
+		public Style Style
 		{
-			_fontElement = new BindableSpan(this);
+			get { return (Style)GetValue(StyleProperty); }
+			set { SetValue(StyleProperty, value); }
 		}
 
+		public static readonly BindableProperty BackgroundColorProperty
+			= BindableProperty.Create(nameof(BackgroundColor), typeof(Color), typeof(Span), default(Color));
+
 		public Color BackgroundColor
 		{
-			get { return _backgroundColor; }
-			set
-			{
-				if (_backgroundColor == value)
-					return;
-				_backgroundColor = value;
-				OnPropertyChanged();
-			}
+			get { return (Color)GetValue(BackgroundColorProperty); }
+			set { SetValue(BackgroundColorProperty, value); }
 		}
 
-		[Obsolete("Font is obsolete as of version 1.3.0. Please use the Font properties directly.")]
-		public Font Font {
-			get { return (Font)_fontElement.GetValue(FontElement.FontProperty); }
-			set { _fontElement.SetValue(FontElement.FontProperty, value); }
+		[Obsolete("Foreground is obsolete as of version 2.6.0. Please use the TextColor property instead.")]
```

Comment:
Since we need to version when this goes obsolete, took a stab at 2.6.0, but of course I don't actually know when this is going out.